File: Data Structures & Algorithms/find-target-in-rotated-sorted-array/submission-0.py
class Solution:
    def search(self, nums: List[int], target: int) -> int:
        l, r = 0, len(nums) - 1

        while l < r:
            m = (l + r) // 2
            if nums[m] > nums[r]:
                l = m + 1
            else:
                r = m

        rot = l
        l, r = 0, len(nums) - 1

        while l <= r:
            m = (l + r) // 2
            realmid = (m + rot) % len(nums)
            if nums[realmid] == target:
                return realmid
            if nums[realmid] < target:
                l = m + 1
            else:
                r = m - 1
        return -1

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.cs
public class Solution {
    public int MinSubArrayLen(int target, int[] nums) {
        int n = nums.Length;
        int res = int.MaxValue;

        for (int i = 0; i < n; i++) {
            int curSum = 0;
            for (int j = i; j < n; j++) {
                curSum += nums[j];
                if (curSum >= target) {
                    res = Math.Min(res, j - i + 1);
                    break;
                }
            }
        }

        return res == int.MaxValue ? 0 : res;
    }
}